SimPEG/EM/NSEM/FieldsNSEM.py

```diff
@@ -0,0 +1,563 @@
+        """
+        Total electric field is sum of primary and secondary
+
+        :param numpy.ndarray solution: field we solved for
+        :param list srcList: list of sources
+        :rtype: numpy.ndarray
+        :return: total electric field
+        """
+        return self._ePrimary(eSolution,srcList) + self._eSecondary(eSolution,srcList)
+
+    def _eDeriv(self, src, v, adjoint = False):
+        """
+        Total derivative of e with respect to the inversion model. Returns :math:`d\mathbf{e}/d\mathbf{m}` for forward and (:math:`d\mathbf{e}/d\mathbf{u}`, :math:`d\mathb{u}/d\mathbf{m}`) for the adjoint
+
+        :param Src src: sorce
+        :param numpy.ndarray du_dm_v: derivative of the solution vector with respect to the model times a vector (is None for adjoint)
+        :param numpy.ndarray v: vector to take sensitivity product with
+        :param bool adjoint: adjoint?
+        :rtype: numpy.ndarray
+        :return: derivative times a vector (or tuple for adjoint)
+        """
+        if getattr(self, '_eDeriv_u', None) is None or getattr(self, '_eDeriv_m', None) is None:
+            raise NotImplementedError ('Getting eDerivs from %s is not implemented' %self.knownFields.keys()[0])
+
+        # if adjoint:
+        #     return self._eDeriv_u(src, v, adjoint), self._eDeriv_m(src, v, adjoint)
+        return np.array(self._eDeriv_u(src, v, adjoint) + self._eDeriv_m(src, v, adjoint), dtype = complex)
+
+    def _eDeriv_u(self, src, v, adjoint = False):
+        """
+        Partial derivative of the total electric field with respect to the solution.
+
+        :param SimPEG.EM.NSEM.Src src: source
+        :param numpy.ndarray v: vector to take product with
+            Size (nE,) when adjoint=True, (nU,) when adjoint=False
+        :param bool adjoint: adjoint?
+        :rtype: numpy.ndarray
+        :return: The calculated derivative, size (nU,) when adjoint=True
+            (nE,) when adjoint=False"""
+
+        return Identity()*v
+
+
+    def _eDeriv_m(self, src, v, adjoint = False):
+        """
+        Partial derivative of the total electric field with respect to the inversion model. Here, we assume that the primary does not depend on the model. Note that this also includes derivative contributions from the sources.
+
+        :param SimPEG.EM.FDEM.Src src: source
+        :param numpy.ndarray v: vector to take product with
+        :param bool adjoint: adjoint?
+        :rtype: SimPEG.Utils.Zero
+        :return: product of the electric field derivative with respect to the inversion model with a vector
+        """
+
+        # assuming primary does not depend on the model
+        return Zero()
+
+    def _bPrimary(self, eSolution, srcList):
+        bPrimary = np.zeros([self.survey.mesh.nE,eSolution.shape[1]], dtype = complex)
+        for i, src in enumerate(srcList):
+            bp = src.bPrimary(self.survey.prob)
+            if bp is not None:
+                bPrimary[:,i] += bp[:,-1]
```

Comment:
It is because the `Source` has both of the polarization's and I am selecting one of them (the py one).